### PR TITLE
allow no packaging, add template used to generated headers

### PR DESCRIPTION
**Pull request by pmcb55**
# New feature description
Not really a new feature, but this change allows YAML files with no packaging info at all, for cases where we just want vocab file generation without packaging up, or publishing.
 
# Checklist

- [X] All acceptance criteria are met.
- [X] Relevant documentation, if any, has been written/updated.
- [X] The changelog has been updated, if applicable.
- [X] New functions/types have been exported in `index.ts`, if applicable.
- [X] Commits in this PR are minimal and [have descriptive commit messages](https://chris.beams.io/posts/git-commit/).

### CODE REVIEW

File: src/generator/ArtifactGenerator.js [hunk 1]
```diff
@@ -336,58 +336,52 @@ class ArtifactGenerator {
...
-
-        // TODO: manage repositories properly
-        this.artifactData.gitRepository = artifactInfo.gitRepository;
-        this.artifactData.repository = artifactInfo.repository;
-        artifactInfo.packaging.forEach((packagingInfo) => {
-          debug(
-            `Generating [${artifactInfo.programmingLanguage}] packaging for [${packagingInfo.packagingTool}]`,
-          );
+        if (artifactInfo.packaging) {
+          // TODO: Manage repositories properly.
```
Comment (NSeydoux) [inline src/generator/ArtifactGenerator.js:340]: Can you create an associated ticket, and reference it in the todo, so that this doesn't just remain here unattended? Or if we are no longer sure what this is about let's just delete the TODO.

File: src/DatasetHandler.js [hunk 2]
```diff
@@ -705,6 +705,7 @@ module.exports = class DatasetHandler {
     result.sourceRdfResources = `Vocabulary built from ${describeInput(
       this.vocabData,
     )}.`;
+    result.sourceTemplateVocab = describeTemplateVocab(this.vocabData);
```
Comment (NSeydoux) [inline src/DatasetHandler.js:708]: Ideally, separate features are added through separate PRs.

**Review by NSeydoux — APPROVED** [review]

File: src/DatasetHandler.js (hunk 2 above)
Comment (pmcb55) [reply src/DatasetHandler.js:708]: Yeah, I know, but I just don't think we have the resource capacity to review multiple PRs in a timely fashion right now (unfortunately), as each review requires a costly 'context switch' for the reviewer.

File: src/generator/ArtifactGenerator.js (hunk 1 above)
Comment (pmcb55) [reply src/generator/ArtifactGenerator.js]: Yeah, pretty sure I didn't add this comment (as it was missing a full-stop!), so I'm not sure what 'properly' means here myself.
So I'll just delete the TODO: 076b3cf
